pinning-services/index.js

```diff
@@ -81,73 +83,121 @@ const job = new cron.CronJob(period, function() {
         }
 
         // Pin API responses to Hypercore
-        if (fs.existsSync(dirApi)) {
-          getDatSeed('dat-seed-api', projName, domain, txtHypercoreApi)
-            .then(seed => hyperPublisher.sync({
-              seed,
-              fsPath: dirApi,
-              drivePath: '/'
-            }))
-            .then(({diff, url}) => {
-              // Log any changes to hyperdrive and return url
-              console.log(`API responses for ${projName} pinned at ${url}. Changes:`);
-              console.log(diff);
-              return url;
-            })
-            .catch(function(error) {
-              console.log(error);
-            });
+        if (protoHypercore) {
+          if (fs.existsSync(dirApi)) {
+            getDatSeed('dat-seed-api', projName, domain, txtHypercoreApi)
+              .then(seed => hyperPublisher.sync({
+                seed,
+                fsPath: dirApi,
+                drivePath: '/'
+              }))
+              .then(({ diff, url }) => {
+                // Log any changes to hyperdrive and return url
+                console.log(`API responses for ${projName} pinned at ${url}. Changes:`);
+                console.log(diff);
+                return url;
+              })
+              .catch(function(error) {
+                console.log(error);
+              });
+          }
         }
 
         // Pin API responses to IPFS
-        if (fs.existsSync(dirApi)) {
-          ipfs.add(globSource(dirApi, { recursive: true, followSymlinks: false }), { cidVersion: 1, pin: true, timeout: 60000 })
-            .then(file => file['cid'])
-            .then(cid => {
-              console.log(`API responses for ${projName} pinned at ipfs/${cid}`);
-
-              // Update DNS record
-              return updateDnsRecordDigitalOcean(domain, 'TXT', txtIpfsApi, `dnslink=/ipfs/${cid}`, 300, conf['digitalOceanAccessToken']);
-            })
-            .catch(function(error) {
-              console.log(error);
-            });
+        if (protoIpfs) {
+          if (fs.existsSync(dirApi)) {
+            ipfs.add(globSource(dirApi, { recursive: true, followSymlinks: false }), { cidVersion: 1, pin: true, timeout: 60000 })
+              .then(file => file['cid'])
+              .then(cid => {
+                console.log(`API responses for ${projName} pinned at ipfs/${cid}`);
+
+                // Update DNS record
+                return updateDnsRecordDigitalOcean(domain, 'TXT', txtIpfsApi, `dnslink=/ipfs/${cid}`, 300, conf['digitalOceanAccessToken']);
+              })
+              .catch(function(error) {
+                console.log(error);
+              });
+          }
+        } else {
+          updateDnsRecordDigitalOcean(domain, 'TXT', txtIpfsApi, '', 300, conf['digitalOceanAccessToken']);
+        }
+
+        // Update HTTP A and AAAA record
+        if (protoHttpDnsUpdate) {
```

Comment:
Why isn't this just `protoHttp`?

Reply:
Please see reference to  https://github.com/hyphacoop/api.distributed.press/pull/3#pullrequestreview-588038687 for config.

The script was updated to REMOVE dns records that are not being used.  For example if IPFS publishingis disable it will remove IPFS TXT records.  Similarly if HTTP is disabled it will remove A and AAAA records.

The removal happens by use of the inline if against `protoHttp`
`updateDnsRecordDigitalOcean(domain, 'AAAA', '@', protoHttp ? `${httpPublisherIpv6Address}` : '', 300, conf['digitalOceanAccessToken']);`

if `updateDnsRecordDigitalOcean` receives a empty string for data it will remove the entry

Wrapping this code inside a conditional protoHttpDnsUpdate  prevents the deletion of the records when http publishing is disabled (or updating records if enabled)

(i would imagine protohttp should be read by the job that updates the nginx script)

Reply:
If this case, you want to have a bool variable called `purgeIfDisabled` for all 3 protocols, and handle them the same in all cases.

Reply:
added purgeIfDisabled  to all 3 protocols, removed dnsupdate